store/store.go

```diff
@@ -0,0 +1,142 @@
+package store
+
+import (
+	"encoding/binary"
+	"errors"
+	"sync"
+
+	"go.uber.org/multierr"
+
+	"github.com/lazyledger/optimint/hash"
+	"github.com/lazyledger/optimint/types"
+)
+
+var (
+	blockPrefix  = [1]byte{1}
+	indexPrefix  = [1]byte{2}
+	commitPrefix = [1]byte{3}
+)
+
+type DefaultStore struct {
+	db KVStore
+
+	height uint64
+
+	// mtx protects height
+	mtx sync.RWMutex
+}
+
+var _ Store = &DefaultStore{}
+
+func New() Store {
+	return &DefaultStore{db: NewInMemoryKVStore()}
+}
+
+func (s *DefaultStore) Height() uint64 {
+	s.mtx.RLock()
+	defer s.mtx.RUnlock()
+	return s.height
+}
```

Comment:
Some of these method aren't super-obvious, can we get some docstrings for exported methods?

Reply:
Commented on interface level.